src/scripting/lua_unit.cpp

```diff
@@ -597,6 +832,32 @@ static int impl_unit_status_set(lua_State *L)
 	return 0;
 }
 
+/**
+ * List statuses on a unit (__dir metamethod)
+ * This returns all known statuses (regardless of state) plus any currently set to true.
```

Comment:
Does it avoid duplicates when a unit is affected by a known state?

Reply:
This logic adds duplicates, yes. However, the `__dir` machinery in `wesnoth.print_attributes` automatically removes duplicates, so I thought it wasn't worth avoiding it here.